engine/access/rpc/engine.go

```diff
@@ -132,21 +132,21 @@ func NewBuilder(log zerolog.Logger,
 	// wrap the unsecured server with an HTTP proxy server to serve HTTP clients
 	httpServer := NewHTTPServer(unsecureGrpcServer, config.HTTPListenAddr)
 
-	// TODO: when cache size is set to 0, handle case where we do not use cache
+	var cache *lru.Cache
 	cacheSize := config.ConnectionPoolSize
-	if cacheSize == 0 {
-		cacheSize = backend.DefaultConnectionPoolSize
```

Comment:
you may need to change the value of `DefaultConnectionPoolSize` to `0` so we can now disable connection pooling by default